docs/docs/components-and-props.md

```diff
@@ -0,0 +1,248 @@
+5. React DOM efficiently updates the DOM to match `<h1>Hello, Sara</h1>`.
+
+>**Caveat:**
+>
+>Always start component names with a capital letter. For example, `<div />` element represents a DOM tag, but `<Welcome />` represents a component. Also make sure your components are in scope when you are using them.
+
+## Composing Components
+
+Components can refer to other components in their output. This lets us use the same component abstraction for any level of detail. A button, a form, a dialog, a screen: in React apps, all those are commonly expressed as components.
+
+For example, we can create an `App` component that renders `Welcome` many times:
+
+```js{8-10}
+function Welcome(props) {
+  return <h1>Hello, {props.name}</h1>;
+}
+
+function App() {
+  return (
+    <div>
+      <Welcome name="Sara" />
+      <Welcome name="Cahal" />
+      <Welcome name="Edite" />
+    </div>
+  );
+}
+
+const container = document.getElementById('root');
+ReactDOM.render(<App />, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/KgQKPr?editors=0010)
+
+Typically, new React apps have a single `App` component as a root. However, if you integrate it into an existing app, you might start bottom-up with a small component like `Button` and gradually work your way to the top of the view hierarchy.
+
+>**Caveat:**
+>
+>Components must return a single root element. This is why we added a `<div>` to contain all the `<Welcome />` elements.
+
+## Extracting Components
+
+Don't be afraid to split components into smaller components.
+
+For example, consider this `Comment` component:
+
+```js
+function Comment(props) {
+  return (
+    <div className="Comment">
+      <div className="UserInfo">
+        <img className="Avatar"
+             src={props.author.avatarUrl}
+             alt={props.author.name} />
+        <div className="UserInfo-name">
+          {props.author.name}
+        </div>
+      </div>
+      <div className="Comment-text">
+        {props.text}
+      </div>
+      <div className="Comment-date">
+        {formatDate(props.date)}
+      </div>
+    </div>
+  );
+}
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/VKQwEo?editors=0010)
+
+It accepts `author` (an object), `text` (a string), and `date` (a date) as props, and describes a comment on a social media website.
+
+This component can be tricky to change because of all the nesting, and it is also hard to reuse individual parts of it. Let's extract a few components from it.
+
+First, we will extract `Avatar`:
+
+```js{3-5}
+function Avatar(props) {
+  return (
+    <img className="Avatar"
+         src={props.user.avatarUrl}
+         alt={props.user.name} />
+  );
+}
+```
+
+The `Avatar` doesn't need to know that it is being rendered inside a `Comment`. This is why we called its prop more generically: `user` rather than `author`. We recommend naming props from the component's own point of view rather than the context in which it is being used.
```

Comment:
> This is why we called its prop more generically

This could sound like you're calling a function. Perhaps: "This is why we named its prop more generically," or to avoid the awkwardly placed modifier, "This is why we have given it a more generic name."